homeassistant/components/caldav/calendar.py

```diff
@@ -312,6 +312,9 @@ def is_matching(vevent, search):
             and pattern.match(vevent.location.value)
             or hasattr(vevent, "description")
             and pattern.match(vevent.description.value)
+            or hasattr(vevent, "categories")
```

Comment:
I don't think we should extend the search config option feature. I rather think we should deprecate the whole custom calendars option in caldav. We have the calendar trigger that can be used to trigger automations that searches for matching events in a condition.

If we need better GUI options for calendars that should be solved generally for all calendar entities and not per integrating integration.

Reply:
Agree, i think there may still be something here around the calendar helper: https://github.com/home-assistant/architecture/discussions/776 but it needs to be revisited or a little more thought.

We can also consider adding "categories" to the calendar event model since it is part of the rfc.